### PR TITLE
removing dependencies on ggpubr and jtools

### PR DESCRIPTION
**Pull request by FBartos**
requires: https://github.com/jasp-stats/jaspGraphs/pull/5

### CODE REVIEW

File: R/MixedModelsCommon.R [hunk 1]
```diff
@@ -1170,34 +1170,28 @@
     
     # add theme
     if (options$plotsTheme == "JASP") {
```
Comment (vandenman) [inline R/MixedModelsCommon.R:1172]: code looks fine but I'd use a switch statement over a long if else tree

**Review by vandenman — APPROVED** [review]

**Comment by vandenman** [comment]
the unit tests fail btw because of:
```r
Error in if (any(nearOne)) { : missing value where TRUE/FALSE needed
```

**Comment by JorisGoosen** [comment]
@vandenman @FBartos is the previous comment a problem or not?

**Comment by FBartos** [comment]
I was having issues installing jaspTools to test it out. Just trying a special Tim's build. This error messages should no be problem, but two figures fail on Windows and Mac gives somewhat different results :/

**Comment by JorisGoosen** [comment]
> I was having issues installing jaspTools to test it out. Just trying a special Tim's build. This error messages should no be problem, but two figures fail on Windows and Mac gives somewhat different results :/

Well, I'd expect at least one of the systems to work to be able to pass this PR.

**Comment by FBartos** [comment]
ok, I'm giving up. One of the GLMM figures on Windows just fails to match but it works perfectly on my local computer. I tried all possible orders of running the tests but I can't reproduce the error. I even tried adding a seed before the plotting function but nada....

The MacOS fails on basically everything because it produces different results than I get on Windows, which is definitely not great. Not sure what I can do about it either.

**Comment by vandenman** [comment]
I'm back from holidays and can take a look somewhere this week.

**Comment by FBartos** [comment]
thanks Don, it would be very helpful - I spent way too much time trying to fix it but I had no success and no idea what else to try

**Comment by vandenman** [comment]
Okay, so the one thing I figured so far is this:

```r
dat <- read.csv("https://raw.githubusercontent.com/jasp-stats/jasp-desktop/development/Resources/Data%20Sets/debug.csv") # change this path
attr(dat$facExperim, "contrasts") <- "contr.sum"
attr(dat$facGender, "contrasts") <- "contr.sum"
ff <- lmerTest::lmer(formula = contNormal ~ contGamma * contBinom * 
    facExperim * facGender + (contGamma + contBinom + facExperim + 
    facGender | facFive), data = dat)
ff

anova_out_ff <- afex:::lmerTest_anova(ff, ddf = "Satterthwaite", type = "3")
```
On Linux (and also my print statement on the MacOS of GitHub Actions) I get:
```r
Type III Analysis of Variance Table with Satterthwaite's method
                                         Sum Sq Mean Sq NumDF  DenDF F value  Pr(>F)  
contGamma                                0.1222  0.1222     1  4.027  0.1335 0.73321  
contBinom                                0.0797  0.0797     1 61.536  0.0871 0.76886  
facExperim                               2.1744  2.1744     1 17.570  2.3758 0.14104  
facGender                                2.1907  2.1907     1 25.714  2.3936 0.13405  
contGamma:contBinom                      0.3093  0.3093     1 75.902  0.3379 0.56275  
contGamma:facExperim                     4.2300  4.2300     1 56.527  4.6220 0.03586 *
contBinom:facExperim                     2.4039  2.4039     1 74.993  2.6267 0.10928  
contGamma:facGender                      1.2842  1.2842     1 77.070  1.4031 0.23984  
contBinom:facGender                      0.7074  0.7074     1 77.605  0.7729 0.38203  
facExperim:facGender                     1.3458  1.3458     1 74.129  1.4705 0.22911  
contGamma:contBinom:facExperim           3.0219  3.0219     1 78.343  3.3019 0.07302 .
contGamma:contBinom:facGender            1.2441  1.2441     1 78.343  1.3594 0.24718  
contGamma:facExperim:facGender           0.2159  0.2159     1 64.787  0.2359 0.62879  
contBinom:facExperim:facGender           1.7802  1.7802     1 76.373  1.9451 0.16716  
contGamma:contBinom:facExperim:facGender 3.6170  3.6170     1 76.152  3.9521 0.05041 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```
but on windows we get:
```r
Type III Analysis of Variance Table with Satterthwaite's method
                                         Sum Sq Mean Sq NumDF  DenDF F value  Pr(>F)  
contGamma                                0.1221  0.1221     1  3.989  0.1334 0.73343  
contBinom                                0.0802  0.0802     1 61.472  0.0876 0.76826  
facExperim                               2.1747  2.1747     1 17.600  2.3761 0.14099  
facGender                                2.1949  2.1949     1 25.744  2.3982 0.13368  
contGamma:contBinom                      0.3096  0.3096     1 75.867  0.3382 0.56258  
contGamma:facExperim                     4.2257  4.2257     1 56.268  4.6171 0.03597 *
contBinom:facExperim                     2.4042  2.4042     1 74.954  2.6268 0.10927  
contGamma:facGender                      1.2911  1.2911     1 76.979  1.4106 0.23860  
contBinom:facGender                      0.7081  0.7081     1 77.572  0.7737 0.38181  
facExperim:facGender                     1.3462  1.3462     1 73.643  1.4708 0.22909  
contGamma:contBinom:facExperim           3.0217  3.0217     1 78.187  3.3015 0.07305 .
contGamma:contBinom:facGender            1.2460  1.2460     1 78.120  1.3614 0.24685  
contGamma:facExperim:facGender           0.2166  0.2166     1 63.819  0.2367 0.62828  
contBinom:facExperim:facGender           1.7804  1.7804     1 76.318  1.9453 0.16714  
contGamma:contBinom:facExperim:facGender 3.6163  3.6163     1 76.133  3.9512 0.05043 .
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ‘ ’ 1
```
The degrees of freedom in the first row are different! It's unclear to me why though.

**Comment by vandenman** [comment]
okay, so the differences between OSes definitely have something to do with there being a singular solution. There's also a 4-way interaction effect in the unit tests... Shouldn't something like
```r
contNormal ~ contGamma * contBinom + (contGamma + contBinom | facFive)
```
suffice? Right now we're doing
```r
contNormal ~ contGamma * contBinom * facExperim * facGender + 
  (contGamma + contBinom + facExperim + facGender | facFive)
```
which might be overkill for testing. This also introduces tons of numerical issues because the debug dataset isn't really suited to estimate complex but also identifiable models.

@FBartos any thoughts?

**Comment by FBartos** [comment]
I took care of the singular fits, but it didn't solve all the problems. I'm officially and completely giving up. This is how I dealt with the remaining problems:

- simplifying overly complex models resulting in singular fits
- rounding estimates at some places to 3 decimal values to solve the remaining differences between windows and MacOS (most likely due to the numerical imprecision)
- skipping one unit test on MacOS and Linux since there were slightly different values in all tables (but I want to keep the windows test)

**Pull request merged**